xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-notifications/src/main/java/org/xwiki/mentions/internal/descriptors/NotificationAdministrationDocumentInitializer.java

```diff
@@ -92,13 +99,20 @@ public boolean updateDocument(XWikiDocument document)
         }
 
         if (!mentionEventPreferenceAlreadySaved) {
+            String wikiId = document.getDocumentReference().getWikiReference().getName();
+            // It is possible for several document with the "XWiki.Notifications.Code.NotificationAdministration" 
+            // document reference to be initialized separately. We keep a map of the computed dates so that every 
```

Comment:
> to be initialized separately

Here you mean "separately and in parallel" right? Because if it's called twice sequentially it should not be a problem as there is a check if the xobject is already present or not.

Reply:
It is a problem even sequentially if two `XWikiDocument` objects are instantiated with the same document reference. In this case they will be initialized with different `startDate`. This is usually not an issue since document initializers usually use static content (so no conflict in the end). But, it is not true here since a new date is instantiated for each call.